primitives/ethereum-primitives/src/merkle.rs

```diff
@@ -0,0 +1,72 @@
+use array_bytes::{array_unchecked, fixed_hex_bytes_unchecked};
+use codec::{Decode, Encode};
+pub use ethereum_types::{H128, H512};
+use sp_io::hashing::sha2_256;
+use sp_runtime::RuntimeDebug;
+use sp_std::vec::Vec;
+
+#[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+pub struct DoubleNodeWithMerkleProof {
```

Comment:
This file is related to ethash proof, it would be better to rename the filename to indicated "ethash"